packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -210,21 +211,45 @@ export class ProjectDataSource {
     }
   }
 
-  async findSpecs (
-    projectRoot: string,
-    testingType: Cypress.TestingType,
-    specPattern: string[],
-    excludeSpecPattern: string[],
-    globToRemove: string[],
-  ): Promise<FoundSpec[]> {
-    const specAbsolutePaths = await this.ctx.file.getFilesByGlob(
+  async findSpecs ({
```

Comment:
Is all of this info derivable? Seems like the info either comes from the `config` or `ctx` but every caller has to repeat the same steps to grab all this info when we could be doing it here. Not that many places we call this though

Reply:
@ZachJW34 looks into this but it doesn't seem like a good idea. If we try to derive it during the unit test, we need to spin up a full `configManager`, and all the overhead that comes with it. The alternative is a "fake" config manager... But then, what are we accomplishing? It's not really any more coverage than what we have here, but a lot more overhead.

I agree this forwarding of options is not really ideal, though. The main problem is we do not normalise the specPatterns from `config` internally, so we need to check if it's either `string | string[] | undefined` everywhere. Ultimately, if we can rework the config layer to normalise everything, it'll be a lot cleaner. Eg - anything that is defined as `string | string[] | undefined` needs to become an array (empty representing `undefined`).

We also should have a full config we can use for unit tests - but at that point, I'd argue an e2e or integration test is better, anyway, like you suggested below.